misc/services/life_mngr/list.h

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (C) 2023 Intel Corporation.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <sys/queue.h>
+
+#define list_foreach_safe(var, head, field, tvar)	\
+for ((var) = LIST_FIRST((head));			\
+	(var) && ((tvar) = LIST_NEXT((var), field), 1); \
+	(var) = (tvar))
```

Comment:
Can we use LIST_FOREACH_SAFE directly?

Reply:
No, glibc does not provide `LIST_FOREACH_SAFE` macro.

> LIST_FOREACH_SAFE(), which is present on the BSDs but is not
> present in glibc, fixes this limitation by allowing var to safely
> be removed from the list and freed from within the loop without
> interfering with the traversal.
> https://man7.org/linux/man-pages/man3/list.3.html